zhusuan/model/base.py

```diff
@@ -115,9 +115,16 @@ def tensor(self):
                         "with its observed value. Error message: {}".format(
                             self._name, e))
             else:
-                self._tensor = self.sample(self._n_samples)
+                self._tensor, self._local_log_prob = self.sample_and_log_prob(self._n_samples)
         return self._tensor
 
+    @property
+    def local_log_prob(self):
+        tensor = self.tensor
+        if not hasattr(self, '_local_log_prob'):
+            self._local_log_prob = self.log_prob(tensor)
+        return self._local_log_prob
```

Comment:
Nice job. Meanwhile can you remove the `log_prob(given)` function in `StochasticTensor`? We should make `StochasticTensor` more static and leave these functions to `StochasticTensor.distribution.log_prob(given)`